File: Chatbot Angular/Chatbot Mobile Version/Chat/src/app/classes/Rule.ts
import { SequenceMessage } from "./SequenceMessage";
import { Trigger } from "./Trigger";
export class Rule{

    constructor(
        public id?,
        public title?: string,
        public description?: string,
        public ruleType?:string,
        public triggers?:Trigger[],
        public sequenceMessages?:SequenceMessage[],
        ){}
}

File: Chatbot WCF/wcf 2/Chatbot/Models/Rule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Runtime.Serialization;

namespace Chatbot.Models
{
    [DataContract]
    [KnownType(typeof(Conversation)), KnownType(typeof(Task))]
    public class Rule
    {
        [DataMember]
        public int id = default(int);
        [DataMember]
        public string title { get; set; }
        [DataMember]
        public string description { get; set; }
        [DataMember]
        public List<Trigger> triggers { get; set; } = new List<Trigger>();
        [DataMember]
        public string ruleType { get; set; }
    }
}